src/modules/endpoint/sagas/fetchResourceCache.js

```diff
@@ -25,10 +25,9 @@ const createErrorMessage = (fetchResult: FetchResultType) => {
   }, '')
 }
 
-export default function * fetchResourceCache (city: string, language: string, fetchMap: FetchMapType, database: MemoryDatabase): Saga<void> {
-  yield call(database.readResourceCache)
-
-  if (isEmpty(fetchMap)) {
+export default function * fetchResourceCache (
+  city: string, language: string, fetchMap: FetchMapType, dataContainer: DataContainer): Saga<void> {
+  if (isEmpty(fetchMap) && dataContainer.resourceCacheLoaded()) {
```

Comment:
Also, why should we handle the fetchMap-empty case separately ? May not be part of this task, but I think this whole if block can be removed... Is this only because we haven't implement the native part on iOS @maxammann? Then we could add a comment here maybe...

But why do you need to check here if resourceCache has loaded?

Reply:
In general if the fetch map is empty we do not have to run the code below. But if the resourceCache in dataContainer is still `null` we want it to be initialized as `{}`.

Reply:
Also if there is nothing to fetch there is no need to run any native code. In theory this affects the performance although I don't know to which extent.